meson.build

```diff
@@ -1460,8 +1460,20 @@ videotoolbox_gl = get_option('videotoolbox-gl').require(
     error_message: 'gl-cocoa nor ios-gl could be found!',
 )
 features += {'videotoolbox-gl': videotoolbox_gl.allowed()}
+corevideo = dependency('appleframeworks', modules: ['CoreVideo'], required: get_option('videotoolbox-pl'))
+videotoolbox_pl = get_option('videotoolbox-pl').require(
+    features['vulkan'] and corevideo.found(),
+    error_message: 'vulkan or CV metal support could be found!',
+)
+features += {'videotoolbox-pl': videotoolbox_pl.allowed()}
+if features['videotoolbox-gl'] or features['videotoolbox-pl']
+    sources += files('video/out/hwdec/hwdec_vt.c')
+endif
 if features['videotoolbox-gl']
-    sources += files('video/out/opengl/hwdec_osx.c')
+    sources += files('video/out/hwdec/hwdec_mac_gl.c')
+endif
+if features['videotoolbox-pl']
+    sources += files('video/out/hwdec/hwdec_vt_pl.m')
```

Comment:
Oh wait sorry I just realized this right after approving. Should you add `corevideo` to the `dependencies` list?